packages/harpguru-core/src/components/activity-legend/activity-legend.tsx

```diff
@@ -0,0 +1,130 @@
+import { useGlobal } from 'reactn'
+import { useTimingTransition } from 'react-native-redash'
+import Animated, {
+  Easing,
+  interpolate,
+  multiply,
+} from 'react-native-reanimated'
+import { View, StyleSheet } from 'react-native'
+import React from 'react'
+import { getPitchIds, getDegreeIds } from 'harpstrata'
+
+import { NoteDisplayFragment } from '../note-display-fragment'
+import { getDisplayValueTuple } from '../../utils'
+import type { DisplayValues } from '../../utils'
+import { DisplayModes } from '../../types'
+import { getSizes, colors } from '../../styles'
+
+const { degreeColors, pageColor, inertOutline } = colors
+
+export const ActivityLegend = (): React.ReactElement => {
+  const [activeHarpStrata] = useGlobal('activeHarpStrata')
+  const [activeDisplayMode] = useGlobal('activeDisplayMode')
+  const {
+    rootPitchId,
+    isActiveComplex: { activePitchIds },
+  } = activeHarpStrata
+  const orderedPitchIds = getPitchIds(rootPitchId)
+  const orderedDegreeIds = getDegreeIds()
+  const sizes = getSizes()
+  const { 9: legendWidth } = sizes
+
+  const activityCells = orderedPitchIds.map((pitchId, index) => {
+    const isActive = activePitchIds.indexOf(pitchId) > -1
+    const { [index]: degreeId } = orderedDegreeIds
+    const { [degreeId]: degreeColor } = degreeColors
+    const legendDisplayMode =
+      activeDisplayMode === DisplayModes.Pitch
+        ? DisplayModes.Degree
+        : DisplayModes.Pitch
+    const displayValue = getDisplayValueTuple(
+      degreeId,
+      pitchId,
+      legendDisplayMode
+    )
+    return (
+      <ActivityCell
+        key={index}
+        isActive={isActive}
+        degreeColor={degreeColor}
+        displayValue={displayValue}
+      />
+    )
+  })
+
+  const styles = StyleSheet.create({
+    block: {
+      ...StyleSheet.absoluteFillObject,
+      width: legendWidth,
+      flexDirection: 'column',
+    },
+  })
+
+  return <View style={styles.block}>{activityCells}</View>
+}
+
+type ActivityCellProps = {
+  readonly degreeColor: string
+  readonly displayValue: DisplayValues
+  readonly isActive: boolean
+}
+
+const ActivityCell = ({
+  degreeColor,
+  displayValue,
+  isActive,
+}: ActivityCellProps): React.ReactElement => {
+  const sizes = getSizes()
+  const { 7: pitchValueSize, 9: legendWidth } = sizes
+
+  const styles = StyleSheet.create({
+    cell: {
+      flex: 1,
+    },
+    cellColor: {
+      ...StyleSheet.absoluteFillObject,
+      backgroundColor: degreeColor,
+    },
+    pitchValuetWrapper: {
```

Comment:
I think we can do without this wrapper if the `cell` style is given a `flexDirection` with all the alignment bits and pieces included too.

Reply:
Done